### DISCUSSION TITLE
Generating MathML output (eventually for MS Word)

### DISCUSSION THREAD

**Discussion by tvercaut** [bdtc/lwarp#25] [Q&A]
I am interested in converting the html generated with `lwarp` to a Microsoft Word (`docx`) document (or rather, I am interested in converting latex to MS Word). This can in theory be done by simply opening the html document in MS Word and then saving it as `docx` or using say `pandoc` to do the conversion.

Unfortunately, the equations get transformed in "raw latex" in the process (this is using the `mathjax` option). I tried using `\booltrue{FormatWP}` but this didn't solve the equation problem.

When using `tex4ht` ot `latexml` instead of `lwarp` (both of which have other issues compared to `lwarp`), equations are transformed in mathml within the html and these are converted well to MS Word.

Looking at the mathjax equations, there is a right click to show mathml so there must be a standard way to convert to mathml. I couldn't find a command line tool for it though. Is it possible for `lwarp` to directly output mathml? Or is there a well known tool taht I missed to do this?

P.S.: Sorry for the multiple questions in a few days.
P.P.S.: For the record, my tex4ht+pandoc conversion script can be found here: https://github.com/cai4cai/XCLThesisTemplate/blob/main/tools/convertlatex2word

**Comment by bdtc** [comment]
If you don't mind SVG images you could try FormatWP but without MathJax.  In fact I should ensure that MathJax is off if FormatWP is on, or at least issue a warning.  I don't think I ever tried it with both on at the same time.

For command-line tools, check:
[LaTeX to MathML conversion: tools, limitations, and approaches?](https://tex.stackexchange.com/questions/227195/latex-to-mathml-conversion-tools-limitations-and-approaches)

**Reply by tvercaut** [reply]
Thanks. I was hoping to be able to get proper equations in MS Word and found pandoc to be limited in its ability to process latex directly. Hence my attempt to produce html with lwarp first and then get from html to docx with pandoc. Pandoc indeed seems to have some support for mathjax but it doesn't seem to work with an html file produced with lwarp. Could it be due to a non-standard use of mathjax in lwarp or is it more likely to be an issue with how pandoc parses the output of lwarp?

**Reply by tvercaut** [reply]
The `tex2chtml-page` command line tool from [MathJax-demos-node](https://github.com/mathjax/MathJax-demos-node/) seems to do mostly what I wanted. However the equation numbers disappear in the output html from `tex2chtml-page`. And when I use this as input to `pandoc`, the generated docx file contains `Undefined control sequence \seteqnumber` lines.

**Reply by tvercaut** [reply]
For the record, I had a related discussion on the pandoc side: https://github.com/jgm/pandoc/discussions/9615
I also found that there is a similar option to `tex2chtml-page` called [`mjcli`](https://github.com/michal-h21/mjcli) used in [`make4ht`](https://github.com/michal-h21/make4ht).
